### PR TITLE
WTBO-4082

### PR DESCRIPTION
**Pull request by PS-SamuelPolavarapu**
Please review for Influx Logging Validation for Commission Junction

### CODE REVIEW

File: Importers/CJ.cs [hunk 1]
```diff
@@ -266,6 +266,19 @@ protected override void DoLoad(DateTime startDate, DateTime endDate, int account
...
+                logger.logValidationFailure(
+                "failed_import_validation",
+                GetAffiliateProgramId().ToString(),
+                this.GetType().ToString(),
+                e.Message.Replace("\"", " "),
+                e.InnerException != null ? e.InnerException.Message : "n/a",
+                orderId,
+                rec.saleAmountUsd ?? "n/a",
+                rec.items != null ? rec.items.Count.ToString() : "n/a",
+                rec.saleAmountUsd != null && rec.items.Count > 0 ? calculateUnitPrice(rec.saleAmountUsd, rec.items.Count.ToString()) : "n/a",
```
Comment (ps-michaelbulotano) [inline Importers/CJ.cs]: The unit price isn't necessarily always going to be the total order amount divided by the quantity of items. Sometimes, if the order amount is $20, one of the items can be $15, and the other can be $5.

You'll need to find a way in which, if the importer goes into the itemDetails loop on line 240 and fails one of the validation there, we report the specific item that the validation failed on.

If the validation failed outside of the itemDetails loop, then I would suggest "n/a" as the Unit Price, as we don't have a specific item to report on (just the order in general).

File: Importers/CJ.cs [hunk 2]
```diff
@@ -451,6 +464,15 @@ private IEnumerable<Sale> CreateSale(Record rec, Redirect redirect)
                 }};
         }
 
+        private string calculateUnitPrice(string saleAmount, string quantity)
```
Comment (PS-Qschmick) [inline Importers/CJ.cs]: Looks like this is duplicated over at https://github.com/PriceSpider-NeuIntel/Sales-Importer/pull/9/files
Is there a central helper class that could contain this `calculateUnitPrice` functionality?

File: Importers/CJ.cs (hunk 1 above)
Comment (PS-SamuelPolavarapu) [reply Importers/CJ.cs]: Resolved.

File: Importers/CJ.cs (hunk 2 above)
Comment (PS-SamuelPolavarapu) [reply Importers/CJ.cs]: We no longer need CalculateUnitPrice functionality in this PR

File: Importers/CJ.cs [hunk 3]
```diff
@@ -251,21 +264,23 @@ protected override void DoLoad(DateTime startDate, DateTime endDate, int account
...
             {
+                logger.logValidationFailure(
+                "failed_import_validation",
+                GetAffiliateProgramId().ToString(),
+                this.GetType().ToString(),
+                e.Message.Replace("\"", " "),
+                e.InnerException != null ? e.InnerException.Message : "n/a",
+                orderId,
+                rec.saleAmountUsd ?? "n/a",
+                rec.items != null ? rec.items.Count.ToString() : "n/a",
```
Comment (ps-michaelbulotano) [inline Importers/CJ.cs]: Should return item.quantity, not the number of unique items in the order.

Comment (PS-SamuelPolavarapu) [reply Importers/CJ.cs]: updated..

**Pull request merged**